extensions/wikia/MyHome/SpecialWikiActivity.class.php

```diff
@@ -9,150 +9,187 @@ class SpecialWikiActivity extends UnlistedSpecialPage {
+
 			return;
 		}
 
 		// choose default view (RT #68074)
-		if ($wgUser->isLoggedIn()) {
+		if ( $wgUser->isLoggedIn() ) {
 			$this->defaultView = MyHome::getDefaultView();
-			if ($par == '') {
+			if ( $par == '' ) {
 				$par = $this->defaultView;
 			}
-		}
-		else {
+		} else {
 			$this->defaultView = false;
 		}
 
 		// watchlist feed
-		if($par == 'watchlist') {
+		if ( $par == 'watchlist' ) {
 			$this->classWatchlist = "selected";
 
 			// not available for anons
-			if($wgUser->isAnon()) {
-				if (get_class(RequestContext::getMain()->getSkin()) == 'SkinOasis') {
-					$wgOut->wrapWikiMsg( '<div class="latest-activity-watchlist-login" >$1</div>', array('oasis-activity-watchlist-login', wfGetReturntoParam()) );
-				}
-				else {
-					$wgOut->wrapWikiMsg( '<div id="myhome-log-in">$1</div>', array('myhome-log-in', wfGetReturntoParam()) );
+			if ( $wgUser->isAnon() ) {
+				if ( get_class( RequestContext::getMain()->getSkin() ) == 'SkinOasis' ) {
+					$wgOut->wrapWikiMsg( '<div class="latest-activity-watchlist-login" >$1</div>', [ 'oasis-activity-watchlist-login', wfGetReturntoParam() ] );
+				} else {
+					$wgOut->wrapWikiMsg( '<div id="myhome-log-in">$1</div>', [ 'myhome-log-in', wfGetReturntoParam() ] );
 				}
 
 				//oasis-activity-watchlist-login
 				// RT #23970
-				$wgOut->addInlineScript(<<<JS
+				$wgOut->addInlineScript( <<<JS
 $(function() {
 	$('#myhome-log-in').find('a').click(function(ev) {
 		openLogin(ev);
 	});
 });
 JS
 				);
-				wfProfileOut(__METHOD__);
+				wfProfileOut( __METHOD__ );
+
 				return;
-			}
-			else {
+			} else {
 				$this->feedSelected = 'watchlist';
 				$feedProxy = new WatchlistFeedAPIProxy();
-				$feedRenderer = new WatchlistFeedRenderer();
 			}
 		} else {
-		//for example: wiki-domain.com/wiki/Special:WikiActivity
+			//for example: wiki-domain.com/wiki/Special:WikiActivity
 			$this->feedSelected = 'activity';
 			$feedProxy = new ActivityFeedAPIProxy();
-			$feedRenderer = new ActivityFeedRenderer();
 		}
 
-		$feedProvider = new DataFeedProvider($feedProxy);
+		$feedProvider = new DataFeedProvider( $feedProxy );
 
 		global $wgJsMimeType, $wgExtensionsPath;
-		$wgOut->addScript("<script type=\"{$wgJsMimeType}\" src=\"{$wgExtensionsPath}/wikia/MyHome/WikiActivity.js\"></script>\n");
-		$wgOut->addExtensionStyle(AssetsManager::getInstance()->getSassCommonURL('extensions/wikia/MyHome/oasis.scss'));
+		$wgOut->addScript( "<script type=\"{$wgJsMimeType}\" src=\"{$wgExtensionsPath}/wikia/MyHome/WikiActivity.js\"></script>\n" );
+		$wgOut->addExtensionStyle( AssetsManager::getInstance()->getSassCommonURL( 'extensions/wikia/MyHome/oasis.scss' ) );
 
-		wfRunHooks( 'SpecialWikiActivityExecute', array( $wgOut, $wgUser ));
+		wfRunHooks( 'SpecialWikiActivityExecute', [ $wgOut, $wgUser ] );
 
-		$data = $feedProvider->get(50);  // this breaks when set to 60...
+		$data = $feedProvider->get( 50 );  // this breaks when set to 60...
 
 		// FIXME: do it in AchievementsII extension
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this "FIXME" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)